compose/cli/errors.py

```diff
@@ -55,6 +56,8 @@ def handle_connection_errors(client):
             if is_ubuntu():
                 exit_with_error(docker_not_found_ubuntu)
             exit_with_error(docker_not_found_generic)
+        if is_docker_for_mac_installed():
+            exit_with_error(conn_error_docker_for_mac)
```

Comment:
It does include a docker binary in the path, so putting it after produces the correct behaviour:

 * If Docker for Mac is not installed, there is no binary in the path, so it'll show an error telling you to install it.
 * If Docker for Mac is installed and not running, it'll show a message telling you to start it.